src/Storage.ts

```diff
@@ -20,7 +20,7 @@ export default class Storage<T> {
   }
 
   async write(data: PersistedData<T>): Promise<void> {
-    await this.storage.setItem(this.key, data);
+    await this.storage.setItem(this.key, data.toString());
```

Comment:
Sole code change here 👀.

`data` here is of type `PersistedData<T> = string | T`, giving us two options:

1. In the case it's `string` then it has a `toString` method, and calling this leaves the string unchanged (_no-op_).
2. In the case it's `T` then this seems to always be inferred as `NormalizedCacheObject`, which is an object whose `toString` would yield `[object Object]`, but where relying on implicit type conversion within `localStorage` would (I believe) give the same result of `[object Object]` (also _no-op_?).